sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/aad/implementation/oauth2/JacksonHttpSessionOAuth2AuthorizedClientRepository.java

```diff
@@ -48,18 +49,19 @@ public void saveAuthorizedClient(OAuth2AuthorizedClient authorizedClient, Authen
         Assert.notNull(authorizedClient, "authorizedClient cannot be null");
         Assert.notNull(request, MSG_REQUEST_CANNOT_BE_NULL);
         Assert.notNull(response, "response cannot be null");
-        Map<String, OAuth2AuthorizedClient> authorizedClients = this.getAuthorizedClients(request);
```

Comment:
what's the point here?

Reply:
I think  we can update the method getAuthorizedClients() directly since it's private:
1. return optional 
2. or  return a  HashMap container ?

Reply:
Please refer to the PR's description. 
In `saveAuthorizedClient`, assume the returned value from `getAuthorizedClients` is immutable will make code more robust.

Reply:
then I have a question, if the map result is not null, so it won't return Collections.emptyMap(), is it still immutable?

<img width="669" alt="image" src="https://user-images.githubusercontent.com/4465723/195055581-0837b49d-bed1-49ea-a1a9-d042277c53f7.png">

Reply:
I agree with @backwind1233, seems like the behaviors of `getAuthorizedClients` are not consistent when session is null or not.

Reply:
It's `SerializerUtils::deserializeOAuth2AuthorizedClientMap`'s responsibility to make sure the returned value is immutable.

Now there are many methods:
1. JacksonHttpSessionOAuth2AuthorizedClientRepository#saveAuthorizedClient
2. JacksonHttpSessionOAuth2AuthorizedClientRepository#getAuthorizedClients
3. SerializerUtils#deserializeOAuth2AuthorizedClientMap

Every method follows `immutable by default` will make code more robust:
1. When pass an object to another method (by return value or method parameter), make it immutable.
2. When get an object from another method, assume it is immutable. If current method needs to modify it, create a new object instead of modifying the object got from other method.